Can't compile Godot3 on windows

Can't compile Godot3 using mingw64:

```
[Initial build] /usr/bin/x86_64-w64-mingw32-g++ -o thirdparty/thekla_atlas/nvmath/ConvexHull.windows.opt.tools.64.o -c -std=c++11 -g1 -O2 -DDEBUG_ENABLED -DWINDOWS_ENABLED -mwindows -DOPENGL_ENABLED -DRTAUDIO_ENABLED -DWASAPI_ENABLED -DWINVER=0x0601 -D_WIN32_WINNT=0x0601 -w -DZSTD_STATIC_LINKING_ONLY -DNV_OS_MINGW -DNV_CC_GNUC -DPOSH_COMPILER_GCC -fopenmp -DMINGW_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_BULLET_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -DMODULE_GRIDMAP_ENABLED -DMODULE_HDR_ENABLED -DMODULE_JPG_ENABLED -DMODULE_MOBILE_VR_ENABLED -DMODULE_OGG_ENABLED -DMODULE_OPENSSL_ENABLED -DMODULE_OPUS_ENABLED -DMODULE_PBM_ENABLED -DMODULE_PVR_ENABLED -DMODULE_RECAST_ENABLED -DMODULE_REGEX_ENABLED -DMODULE_SQUISH_ENABLED -DMODULE_STB_VORBIS_ENABLED -DMODULE_SVG_ENABLED -DMODULE_TGA_ENABLED -DMODULE_THEKLA_UNWRAP_ENABLED -Icore -Icore/math -Ieditor -Idrivers -I. -Iplatform/windows -Ithirdparty/zstd -Ithirdparty/zstd/common -Ithirdparty/zlib -Ithirdparty/rtaudio -Ithirdparty/glad -Ithirdparty/thekla_atlas -Ithirdparty/thekla_atlas/poshlib -Ithirdparty/thekla_atlas/nvcore -Ithirdparty/thekla_atlas/nvmesh thirdparty/thekla_atlas/nvmath/ConvexHull.cpp
In file included from thirdparty/thekla_atlas/nvmath/Basis.h:7:0,
                 from thirdparty/thekla_atlas/nvmath/Basis.cpp:3:
thirdparty/thekla_atlas/nvmath/nvmath.h: In function 'bool nv::isFinite(float)':
thirdparty/thekla_atlas/nvmath/nvmath.h:174:16: error: '_finite' was not declared in this scope
         return _finite(f) != 0;
                ^~~~~~~
thirdparty/thekla_atlas/nvmath/nvmath.h:174:16: note: suggested alternative: '_finitef'
         return _finite(f) != 0;
                ^~~~~~~
                _finitef
thirdparty/thekla_atlas/nvmath/nvmath.h: In function 'bool nv::isNan(float)':
thirdparty/thekla_atlas/nvmath/nvmath.h:189:16: error: '_isnan' was not declared in this scope
         return _isnan(f) != 0;
                ^~~~~~
thirdparty/thekla_atlas/nvmath/nvmath.h:189:16: note: suggested alternative: '_isnanf'
         return _isnan(f) != 0;
                ^~~~~~
                _isnanf
scons: *** [thirdparty/thekla_atlas/nvmath/Basis.windows.opt.tools.64.o] Error 1
In file included from thirdparty/thekla_atlas/nvmath/ConvexHull.h:7:0,
                 from thirdparty/thekla_atlas/nvmath/ConvexHull.cpp:3:
thirdparty/thekla_atlas/nvmath/nvmath.h: In function 'bool nv::isFinite(float)':
thirdparty/thekla_atlas/nvmath/nvmath.h:174:16: error: '_finite' was not declared in this scope
         return _finite(f) != 0;
                ^~~~~~~
thirdparty/thekla_atlas/nvmath/nvmath.h:174:16: note: suggested alternative: '_finitef'
         return _finite(f) != 0;
                ^~~~~~~
                _finitef
thirdparty/thekla_atlas/nvmath/nvmath.h: In function 'bool nv::isNan(float)':
thirdparty/thekla_atlas/nvmath/nvmath.h:189:16: error: '_isnan' was not declared in this scope
         return _isnan(f) != 0;
                ^~~~~~
thirdparty/thekla_atlas/nvmath/nvmath.h:189:16: note: suggested alternative: '_isnanf'
         return _isnan(f) != 0;
                ^~~~~~
                _isnanf
scons: *** [thirdparty/thekla_atlas/nvmath/ConvexHull.windows.opt.tools.64.o] Error 1
```

Compiled on Digitecnology Server for bringing Godot 3 builds. Commit id: f3ad14224

--- a/modules/thekla_unwrap/SCsub
+++ b/modules/thekla_unwrap/SCsub
@@ -67,7 +67,7 @@ if env['builtin_thekla_atlas']:
         if env.msvc:
             env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_WIN32", "-DNV_CC_MSVC", "-DPOSH_COMPILER_MSVC" ])
         else:
-            env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_MINGW", "-DNV_CC_GNUC", "-DPOSH_COMPILER_GCC"])
+            env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_MINGW", "-DNV_CC_GNUC", "-DPOSH_COMPILER_GCC", "-U__STRICT_ANSI__"])
             env.Append(LIBS=["dbghelp"])
         
 # Godot source files

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -308,6 +308,10 @@ Files extracted from the upstream source:
 - Relevant sources from src/
 - License.txt
 
+Important: Some files have Godot-made changes, those
+changes are marked with `// -- GODOT --` comments.
+
+
 ## nanosvg
 
 - Upstream: https://github.com/memononen/nanosvg

--- a/thirdparty/thekla_atlas/nvcore/Debug.cpp
+++ b/thirdparty/thekla_atlas/nvcore/Debug.cpp
@@ -14,6 +14,7 @@
 #   define VC_EXTRALEAN
 #   include <windows.h>
 #   include <direct.h>
+// -- GODOT start -
 #   include <crtdbg.h>
 #   if _MSC_VER < 1300
 #       define DECLSPEC_DEPRECATED
@@ -24,6 +25,7 @@
 // VC7: ships with updated headers
 #       include <dbghelp.h>
 #   endif
+// -- GODOT end -
 #   pragma comment(lib,"dbghelp.lib")
 #endif
 
@@ -107,8 +109,9 @@ namespace
 
 #endif
 
-
+// -- GODOT start -
 #if NV_OS_WIN32 || NV_OS_DURANGO
+// -- GODOT end -
 
     // We should try to simplify the top level filter as much as possible.
     // http://www.nynaeve.net/?p=128

--- a/thirdparty/thekla_atlas/nvcore/DefsGnucWin32.h
+++ b/thirdparty/thekla_atlas/nvcore/DefsGnucWin32.h
@@ -19,7 +19,9 @@
 #endif
 
 #define NV_FASTCALL		__attribute__((fastcall))
+// -- GODOT start -
 #define NV_FORCEINLINE	__attribute__((always_inline)) inline
+// -- GODOT end -
 #define NV_DEPRECATED   __attribute__((deprecated))
 
 #if __GNUC__ > 2

